packages/block-library/src/block/edit.native.js

```diff
@@ -0,0 +1,262 @@
+
+/**
+ * Internal dependencies
+ */
+import styles from './editor.scss';
+import EditTitle from './edit-title';
+
+class ReusableBlockEdit extends Component {
+	constructor( { reusableBlock } ) {
+		super( ...arguments );
+
+		this.setBlocks = this.setBlocks.bind( this );
+		this.toggleSheet = this.toggleSheet.bind( this );
+		this.requestFallback = this.requestFallback.bind( this );
+
+		if ( reusableBlock ) {
+			// Start in edit mode when we're working with a newly created reusable block
+			this.state = {
+				// Since edition is not supported yet isEditing is always false
+				isEditing: false,
+				blocks: parse( reusableBlock.content ),
+				showHelp: false,
+				sendFallbackMessage: false,
+			};
+		} else {
+			// Start in preview mode when we're working with an existing reusable block
+			this.state = {
+				isEditing: false,
+				blocks: [],
+				showHelp: false,
+				sendFallbackMessage: false,
+			};
+		}
+	}
+
+	componentDidMount() {
+		this.props.fetchReusableBlock();
+	}
+
+	componentDidUpdate( prevProps ) {
+		const hasBlockFetchCompleted =
+			! prevProps.reusableBlock && this.props.reusableBlock;
+		const hasBlockContentChanged =
+			prevProps.reusableBlock?.content !==
+			this.props.reusableBlock?.content;
+
+		if ( hasBlockFetchCompleted || hasBlockContentChanged ) {
+			this.setState( {
+				blocks: parse( this.props.reusableBlock.content ),
+			} );
+		}
+	}
+
+	toggleSheet() {
+		this.setState( {
+			showHelp: ! this.state.showHelp,
+		} );
+	}
+
+	requestFallback() {
+		this.toggleSheet();
+		this.setState( { sendFallbackMessage: true } );
+	}
+
+	setBlocks( blocks ) {
+		this.setState( { blocks } );
+	}
+
+	renderSheet() {
+		const { reusableBlock } = this.props;
+		const { showHelp } = this.state;
+
+		const { getStylesFromColorScheme, clientId } = this.props;
+		const infoTextStyle = getStylesFromColorScheme(
+			styles.infoText,
+			styles.infoTextDark
+		);
+		const infoTitleStyle = getStylesFromColorScheme(
+			styles.infoTitle,
+			styles.infoTitleDark
+		);
+		const infoSheetIconStyle = getStylesFromColorScheme(
+			styles.infoSheetIcon,
+			styles.infoSheetIconDark
+		);
+		const actionButtonStyle = getStylesFromColorScheme(
+			styles.actionButton,
+			styles.actionButtonDark
+		);
+
+		const infoTitle =
+			Platform.OS === 'android'
+				? __(
+						"'Reusable blocks aren't editable on WordPress for Android"
+				  )
+				: __( "Reusable blocks aren't editable on WordPress for iOS" );
+
+		return (
+			<BottomSheet
+				isVisible={ showHelp }
+				hideHeader
+				onClose={ this.toggleSheet }
+				onModalHide={ () => {
+					if ( this.state.sendFallbackMessage ) {
+						// On iOS, onModalHide is called when the controller is still part of the hierarchy.
+						// A small delay will ensure that the controller has already been removed.
+						this.timeout = setTimeout( () => {
+							requestUnsupportedBlockFallback(
+								`<!-- wp:block {"ref":${ reusableBlock.id }} /-->`,
```

Comment:
This was the reason why the edition through UBE wasn't working, we need to pass the `wp:block` code and I was passing the reusable block content.

Reply:
Nice find!

I'm wondering if we can get this html snippet somehow without hardcoding it. Its structure can easily change breaking the web experience.

Reply:
We'll also need to incorporate the recent changes from https://github.com/WordPress/gutenberg/pull/25238 that allow WP app users with Single Sign On disabled to enable it from this screen.

Reply:
>I'm wondering if we can get this html snippet somehow without hardcoding it. Its structure can easily change breaking the web experience.

Good point, I'll figure out a way to pass this content from the parent instead of hardcoding it here.

> We'll also need to incorporate the recent changes from #25238 that allow WP app users with Single Sign On disabled to enable it from this screen. It's probably going to be best to make a separate component that handles SSO and use it here.

Ok, I'll review the PR and apply the same changes in this component.